### PR TITLE
Change trust of SupersedesTransform

### PR DESCRIPTION
**Pull request by mlsteele**
In line after https://github.com/keybase/client/pull/6010

Use `GetSuperseders` to make sure that if the server e.g. sets all `SupersededBy` to `0` that we won't miss supersede relations. This only helps if the window of `supersedesTransform` encompasses downwhen enough to get the superseder. I'm not 100% that this is a real issue nor a real fix. Sorry.

cc @oconnor663

### CODE REVIEW

**Comment by mmaxim** [comment]
Why does CI fail?

**Comment by mlsteele** [comment]
KBFS `TestChatResetAccount`. I suspect flake.

**Comment by mlsteele** [comment]
Actually I don't suspect it's a flake, could be any of this PR tower.

**Comment by mlsteele** [comment]
CI fixed

File: go/chat/supersedes.go [hunk 1]
```diff
@@ -86,25 +86,47 @@ func (t *supersedesTransform) run(ctx context.Context,
...
 			if supersededBy > 0 {
-				superMsgIDs = append(superMsgIDs, supersededBy)
+				superMsgIDSet[supersededBy] = true
+			}
+			supers, err := utils.GetSupersedes(msg)
+			if err != nil {
+				continue
+			}
+			if len(supers) > 0 {
+				superMsgIDSet[msg.GetMessageID()] = true
```
Comment (mmaxim) [inline go/chat/supersedes.go]: Shoudln't this set to something from `supers`

Comment (mlsteele) [inline go/chat/supersedes.go]: `supers` is a list of supersed**ed** messages. Here I'm collecting a list of supersed**ers** into `superMsgIDSet`.

**Pull request closed without merging**